dom/src/main/scala/fs2/dom/Window.scala

```diff
@@ -67,7 +72,19 @@ object Window {
 
       implicit def given_Dom_F: Dom[F] = Dom.forAsync
 
-      def document: HtmlDocument[F] = window.document.asInstanceOf[HtmlDocument[F]]
+      def document: HtmlDocument[F] =
+        window.document.asInstanceOf[HtmlDocument[F]]
+
+      def asyncDomContentLoaded: F[Unit] =
+        EventTargetHelpers
+          .listen1[F, dom.Event](window.asInstanceOf[dom.Window], "DOMContentLoaded")
+          .void
+
+      def domContentLoaded: F[Unit] =
+        document.readyState.flatMap {
+          case dom.DocumentReadyState.loading => asyncDomContentLoaded
+          case _                              => F.unit
+        }
```

Comment:
Hmm, this got a lot more complicated, why do we have two methods now?

I liked your old approach, I just suggested to use `asyncCheckAttempt`. It was specifically designed for this situation, where there is a race condition and you need to check something after setting up the callback.

Reply:
I thought that exposing the `readyState` was a fairly useful; during initialization you either want to wait for the `domContentLoaded` or all assets being all loaded, both of which have similar logic (or you go via the other state change handler).

I don't particularly mind hiding one of the methods, and/or switching the implementation to use `asyncCheckAttempt` rather than reusing `EventTargetHelpers`.

Reply:
> I thought that exposing the `readyState` was a fairly useful

I'm 👍to this. What I am confused about is why we have both `domContentLoaded` and `asyncDomContentLoaded`.

Reply:
Oh right, I wondered whether it was useful to expose just the registering of the handler alone as a separate function or not; have removed it if it's not useful. The name was really a placeholder.